plugins/repoquery.py

```diff
@@ -218,35 +222,48 @@ def filter_repo_arch(opts, query):
         return query
 
     def configure(self, args):
+        (self.opts, self.parser) = parse_arguments(args)
+
+        if self.opts.srpm:
+            dnfpluginscore.lib.enable_source_repos(self.base.repos)
+
         demands = self.cli.demands
         demands.sack_activation = True
         demands.available_repos = True
 
     def run(self, args):
-        (opts, parser) = parse_arguments(args)
 
-        if opts.help_cmd:
-            print(parser.format_help())
+        if self.opts.help_cmd:
+            print(self.parser.format_help())
             return
 
-        if opts.querytags:
+        if self.opts.querytags:
             print(_('Available query-tags: use --queryformat ".. %{tag} .."'))
             print(QUERY_TAGS)
             return
 
-        if opts.key:
-            q = dnf.subject.Subject(opts.key, ignore_case=True).get_best_query(
+        if self.opts.key:
+            q = dnf.subject.Subject(self.opts.key, ignore_case=True).get_best_query(
                 self.base.sack, with_provides=False)
         else:
             q = self.base.sack.query()
 
-        if opts.pkgfilter == "duplicated":
+        if self.opts.srpm:
```

Comment:
I am not sure if we want to transfer packages to srpm right after packages are found by main provides/pkgname expresion given or after additional filters (`--whatprovides`, `--whatrequires`, `--file`, `unsatisfied`, ..) application before printing package attributes (`fmt_fn = build_format_fn(self.opts)` line). Opinions @rholy, @MichaelMraka, @bkabrda ?